mobile/Makefile

```diff
@@ -0,0 +1,173 @@
+# build once, which will use Xcode to download the signing profile.
+ios-release: apply-patches
+	tns prepare ios --release
+	rm -rf platforms/ios/build
+	xcodebuild -workspace platforms/ios/mobile.xcworkspace -scheme mobile -destination generic/platform=iOS archive -archivePath `pwd`/platforms/ios/build/mobile.xcarchive
+	xcodebuild -exportArchive -archivePath `pwd`/platforms/ios/build/mobile.xcarchive -exportOptionsPlist app/App_Resources/iOS/ExportOptions.plist -exportPath `pwd`/platforms/ios/build
+	ls -l `pwd`/platforms/ios/build/mobile.ipa
+
+xcode-dev: ios-dev
+	open platforms/ios/mobile.xcworkspace/
+
+release: android-release-32 gradle-prod android-compile gradle-simul android-release-copy ios-prepare
+	open platforms/ios/mobile.xcworkspace/
+
+##
+## Code-related targets
+##
+
+cothority-pull:
+	make -C .. cothority-pull
+
+kyber_src: cothority_src app/lib/kyber
+	npm r @c4dt/kyber
+
+app/lib/kyber: ../cothority/external/js/kyber/src
+	@cp -a $< $@
+	@find app/ -name "*.ts" | xargs perl -pi -e "s:\@c4dt/kyber:~/lib/kyber:"
+	@find $@ -name "*.ts" | grep -v .d.ts | xargs -n 1 ./nodeify.sh
+
+kyber_npm:
+	@echo "Using kyber-npm libraries"
+	@if [ ! -d app/lib/kyber ]; then \
+		echo "there is no kyber-source present, aborting"; \
+		exit 0; \
+	fi; \
+	find app/lib/kyber -name "*.js" | xargs rm ;\
+	find app/lib/kyber -name "*.ts" | grep -v .d.ts | xargs -n 1 ./nodeify.sh remove; \
+	diff -Naurq ../cothority/external/js/kyber/src/ app/lib/kyber/ || \
+		( echo "Moving changes to kyber"; cp -a app/lib/kyber/ ../cothority/external/js/kyber/src ); \
+	rm -rf app/lib/kyber; \
+	find app/ -name "*.ts" | xargs perl -pi -e "s:~/lib/kyber:\@c4dt/kyber:"
+	npm i -s @c4dt/kyber
+
+cothority_src: dynacred_src cothority-pull app/lib/cothority
+	npm r @c4dt/cothority
+
+app/lib/cothority: ../cothority/external/js/cothority/src
+	@cp -a $< $@
+	@find app/ -name "*.ts" | xargs perl -pi -e "s:\@c4dt/cothority:~/lib/cothority:"
+	@find $@ -name "*.ts" | grep -v .d.ts | xargs -n 1 ./nodeify.sh
+
+cothority_npm: kyber_npm
+	@echo "Using cothority-npm libraries"
+	@if [ ! -d app/lib/cothority ]; then \
+		echo "there is no cothority-source present"; \
+		exit 0; \
+	fi; \
+	find app/lib/cothority -name "*.js" | xargs rm ;\
+	find app/lib/cothority -name "*.ts" | grep -v .d.ts | xargs -n 1 ./nodeify.sh remove; \
+	diff -Naurq ../cothority/external/js/cothority/src/ app/lib/cothority/ || \
+		( echo "Moving changes to cothority"; cp -a app/lib/cothority/ ../cothority/external/js/cothority/src ); \
+	rm -rf app/lib/cothority; \
+	find app/ -name "*.ts" | xargs perl -pi -e "s:~/lib/cothority:\@c4dt/cothority:"
+	npm i -s @c4dt/cothority
+
+dynacred_src: app/lib/dynacred
+	npm r @c4dt/dynacred
+
+app/lib/dynacred: ../dynacred/src
+	@cp -a $< $@
+	@find app/ -name "*.ts" | xargs perl -pi -e "s:\@c4dt/dynacred:~/lib/dynacred:"
+	@find $@ -name "*.ts" | grep -v .d.ts | xargs -n 1 ./nodeify.sh
+
+dynacred_npm: cothority_npm
```

Comment:
And that's the first copy-paste block I see; you might want to commonise these, to avoid subtle issues.

Reply:
yes, I don't feel like spending another hour making these three blocks the same and making sure that all escape-chars make it correctly through the make/bash/perl chain...